### ISSUE TITLE
False positive in vue/no-dupe-keys 

### ISSUE THREAD

**Issue by sawmurai** [vuejs/eslint-plugin-vue#2207]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.42.0
- **eslint-plugin-vue version:** 9.14.1
- **Node version:** 18
- **Operating System:** Mac

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true,
    es6: true,
  },
  extends: ['plugin:prettier/recommended'],
  plugins: ['prettier'],
  parserOptions: {
    ecmaVersion: 2020,
    tsconfigRootDir: __dirname,
  },
  rules: {
    // global library settings
    'prettier/prettier': 'error',
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    // package specific rules
    'no-await-in-loop': 'off',
    // global library settings (disable unicorn entirely)
    'unicorn/consistent-destructuring': 'off',
    'unicorn/custom-error-definition': 'off',
    'unicorn/empty-brace-spaces': 'off',
    'unicorn/error-message': 'off',
    'unicorn/escape-case': 'off',
    'unicorn/expiring-todo-comments': 'off',
    'unicorn/new-for-builtins': 'off',
    'unicorn/no-abusive-eslint-disable': 'off',
    'unicorn/no-array-push-push': 'off',
    'unicorn/no-for-loop': 'off',
    'unicorn/no-hex-escape': 'off',
    'unicorn/no-instanceof-array': 'off',
    'unicorn/no-keyword-prefix': 'off',
    'unicorn/no-lonely-if': 'off',
    'unicorn/no-nested-ternary': 'off',
    'unicorn/no-new-array': 'off',
    'unicorn/no-new-buffer': 'off',
    'unicorn/no-object-as-default-parameter': 'off',
    'unicorn/no-process-exit': 'off',
    'unicorn/no-unreadable-array-destructuring': 'off',
    'unicorn/no-unsafe-regex': 'off',
    'unicorn/no-unused-properties': 'off',
    'unicorn/no-zero-fractions': 'off',
    'unicorn/number-literal-case': 'off',
    'unicorn/prefer-add-event-listener': 'off',
    'unicorn/prefer-array-find': 'off',
    'unicorn/prefer-array-flat-map': 'off',
    'unicorn/prefer-array-index-of': 'off',
    'unicorn/prefer-array-some': 'off',
    'unicorn/prefer-date-now': 'off',
    'unicorn/prefer-default-parameters': 'off',
    'unicorn/prefer-dom-node-append': 'off',
    'unicorn/prefer-dom-node-dataset': 'off',
    'unicorn/prefer-dom-node-remove': 'off',
    'unicorn/prefer-dom-node-text-content': 'off',
    'unicorn/prefer-includes': 'off',
    'unicorn/prefer-keyboard-event-key': 'off',
    'unicorn/prefer-modern-dom-apis': 'off',
    'unicorn/prefer-negative-index': 'off',
    'unicorn/prefer-number-properties': 'off',
    'unicorn/prefer-optional-catch-binding': 'off',
    'unicorn/prefer-query-selector': 'off',
    'unicorn/prefer-reflect-apply': 'off',
    'unicorn/prefer-regexp-test': 'off',
    'unicorn/prefer-set-has': 'off',
    'unicorn/prefer-string-replace-all': 'off',
    'unicorn/prefer-string-slice': 'off',
    'unicorn/prefer-string-starts-ends-with': 'off',
    'unicorn/prefer-string-trim-start-end': 'off',
    'unicorn/prefer-ternary': ['off', 'only-single-line'],
    'unicorn/prefer-type-error': 'off',
    'unicorn/throw-new-error': 'off',
  },
};

```

**What did you do?**	

```vue
<script setup lang="ts">
// ...
import {useTableRow} from '../../../composables/table-row';
// ...
const tableRow = useTableRow(props, emits);

// Error in the following line
const {onTableRowClick} = tableRow;
// ...
</script>
```

**What did you expect to happen?**
No error

**What actually happened?**

219:7  error  Duplicated key 'onTableRowClick'  vue/no-dupe-keys

**Repository to reproduce this issue**

Its a private repository, so I cannot share it. Creating a demo repo will only be possible to me after work hours. Since the bug is in the same rule that previously was fixed with 9.14.1, I hope it won't be necessary.

**Comment by sawmurai** [comment]
[Reproduction in playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJx1k8+O0zAQxl9l8B6aSptGsBdI07JouXCCA+Ky4eDETjF1bct2KqooT8GVp+NJGMf5BxKn2vP9ZvzNZNqRWjNOclJ4fjGSen4sFUAhlGk9XNMLqvJQkuH+hcqWlwSyyFSt91rBYy1FfUbmG1VM8qdwQ6iW1DmMUsbSSJbk+I4x+Kzf6yKLoViolTPtNdOpFM4jHDRUpYgHQDuNtiMEQkH4/YC+XUkmJD/z20jsBFvif9UXmLNKmbQONrOcpkwrvsmHN3bhDP2SMre8qAd4MV8m8DjxhTNUoX3RrHOwxd+/fhZZEI/QdfEtz3946Pux+UyKOKOslXgostVnKlxthfHguG8N3sXFaOuhA8sbgB4aqy9Qkms7GKq1ch6W74iOkUtQIttJnQc6is9fUZpEY7UJAuONUPxTuBUdzeFlf0wQg4k7cf+x+o5gsoUD9hXsW7RoFVoL/D1UObzCDvv9Uh0ltHwYs5Ptfq6oJd9JfUpoiOG0hqaH9v1NhjnslhWLr9VaapvjvCuhWBKVpxAbqoLRQnluU37lyruZWy85vIWNUMILKjeQw0aFZcBBYT763s1bGt8Lp3Rwk0Mg11RYpkhVtD6frG4VS0eDd5w3/7Jx8f6b8dBUb14/7Fdd3jXNWARnM06E3BNDreMW/9iP/mZ4HFrKncTWs1Hr/wC30Fha)

the reproduction makes me think that this might actually be correct and not a bug in the plugin ...

**Comment by ota-meshi** [comment]
I believe the rule is working correctly.
Turn off the rule if you don't need it.
